src/strategies/immutable-x/index.ts

```diff
@@ -0,0 +1,138 @@
+        { blockTag: 'latest' }
+    );
+    return mapL1Response(response, addresses, options);
+}
+
+async function getL2Balances(
+    network: string,
+    options: { address: string; decimals: string; pageSize: string },
+    addresses: string[]
+): Promise<Record<string, number>> {
+    const records: Record<string, number> = {};
+
+    let cursor = '',
+        recordsLen = addresses.length; // assume all addresses exist
+
+    while (recordsLen != 0) {
+        const apiUrl = buildURL(network, options, addresses, cursor);
+        const response = await fetch(apiUrl, {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        });
+        const resJson = await response.json();
+        Object.assign(records, mapL2Response(resJson, options));
+        cursor = resJson.cursor;
+        recordsLen = (resJson as Response).records.length;
+    }
+    return records;
+}
+
+function buildURL(
+    network: string,
+    options: { address: string; pageSize: string },
+    addresses: string[],
+    cursor?: string
+): string {
+    let apiUrl = networkMapping[network] + snapshotPath;
+    apiUrl += '/' + options.address.toLowerCase() + '?addresses=' + addresses.join(',');
+    apiUrl += `?page_size=${options.pageSize != '' ? options.pageSize : defaultPageSize}`;
+    apiUrl += cursor || cursor != '' ? `?cursor=${cursor}` : '';
+    return apiUrl;
+}
+
+function mapL1Response(
+    data: BigNumber[][], // itf.decodeFunctionResult('balanceOf', '0x...')
+    addresses: string[],
+    options: { decimals: string }
+): Record<string, number> {
+    return Object.fromEntries(
+        data.map((value: BigNumber[], i: number) => [addresses[i], formatBalance(value[0], options.decimals)])
+    );
+}
+
+function mapL2Response(data: Response, options: { decimals: string }): Record<string, number> {
+    return Object.fromEntries(
+        data.records.map((value: Score) => [value.address, formatBalance(value.balance, options.decimals)])
+    );
+}
+
+function formatBalance(balance: BigNumber | string, decimals: string): number {
+    return parseFloat(formatUnits(balance, decimals));
+}
+
+function combineBalanceScores(records: Record<string, number>[]): Record<string, number> {
+    return records.reduce((aggScore, currScore) => {
+        for (const [address, balance] of Object.entries(currScore)) {
+            if (!aggScore[address]) {
+                aggScore[address] = balance;
+            } else {
+                aggScore[address] += balance; // sum(L1, L2)
+            }
+        }
+        return aggScore;
+    }, {});
+}
+
+export async function strategy(
+    _space,
+    network,
+    provider,
+    addresses,
+    options,
+    _snapshot
+): Promise<Record<string, number>> {
+    try {
+        return combineBalanceScores([
+            await getL1Balances(network, provider, options, addresses),
```

Comment:
The strategy should check balance at a specific block number, but here I don't see any block number (_snapshot is not used), this will make result changing overtime, and doesn't make it really usable. Is Immutable X EVM compatible? Isn't possible to use it with Ethers.js provider?

Reply:
Unfortunately, we don't currently support block (batch on l2)-level snapshots in our system (this will change soon) but current idea is to take a snapshot once based on `latest` right before the proposal is published and not have it change again for the lifetime of the proposal. Is there a way to enforce this?